instapipeline/autocrop.py

```diff
@@ -0,0 +1,184 @@
+"""
+This module contains functions for autocropping.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+from skimage.io import imread
+from sklearn.neighbors import KDTree
+from sklearn.cluster import AffinityPropagation, KMeans
+from scipy import ndimage
+
+# list of declumping algs handled
+declumping_algs = ['KMeans']
+
+# autocrop parameters
+pref_param = -50000
+crosshair_ratio = 0.04
+
+
+def get_nnd(coord, kdt):
+    dist, ind = kdt.query([coord], k=2)
+    return dist[0][1]
+
+
+def get_bb_tuples(coords, crosshair_arm_length, max_num_crops):
```

Comment:
It looks like a lot of the functions don't have the parameters and returned variables documented in their docstrings. At a minimum, can you please go through and document all of the "public ones" (i.e., the ones you would actually call during your analysis). I think it is okay for now if you just describe what the function does for the more "utility" functions (i.e., the functions that aren't directly called by the user, but are called by the "public" functions).

Reply:
Got it. Will do!